### PR TITLE
MBS-10881 build-logic refactorings

### PR DESCRIPTION
**Pull request by dsvoronin**
remove nebulaPluginVersion property duplicate
rename build-logic modules, to exclude verbose "-convention" suffix
create build-logic/checks module
add "convention.detekt"
remove systemprop.kotlinVersion
remove keeper plugin setup
workaround to use version catalog in buildscript and convention plugins

### CODE REVIEW

File: build-logic/kotlin/src/main/kotlin/convention.kotlin-base.gradle.kts [hunk 1]
```diff
@@ -4,15 +4,10 @@ plugins {
...
-    add("implementation", "org.jetbrains.kotlin:kotlin-stdlib:1.5.20")
-}
-
 /**
- * 1.4 used because of kotlin dsl bundled version
- * todo use 1.5
+ * overrides 1.4, that comes from kotlin dsl bundled version
  */
-val kotlinLanguageVersion = "1.4"
+val kotlinLanguageVersion = "1.5"
```
Comment (eugene-krivobokov) [inline build-logic/kotlin/src/main/kotlin/convention.kotlin-base.gradle.kts]: I'd good to make this change in a separate PR.
I don't see how it's related to other changes.

**Review by eugene-krivobokov — APPROVED** [review]

File: build-logic/kotlin/src/main/kotlin/convention.kotlin-base.gradle.kts (hunk 1 above)
Comment (dsvoronin) [reply build-logic/kotlin/src/main/kotlin/convention.kotlin-base.gradle.kts]: i agree, let me separate it

File: build-logic/testing/src/main/kotlin/convention.unit-testing.gradle.kts [hunk 2]
```diff
@@ -1,3 +1,5 @@
+import gradle.kotlin.dsl.accessors._aefff6f539276b188c30d8843d1e61e9.testImplementation
+import gradle.kotlin.dsl.accessors._aefff6f539276b188c30d8843d1e61e9.testRuntimeOnly
```
Comment (eugene-krivobokov) [inline build-logic/testing/src/main/kotlin/convention.unit-testing.gradle.kts:2]: What is this package? Does it expected?

Comment (dsvoronin) [reply build-logic/testing/src/main/kotlin/convention.unit-testing.gradle.kts:2]: nope, it could lead to accessor problems. let me think if we can defent from it via detekt or something similar

Comment (dsvoronin) [reply build-logic/testing/src/main/kotlin/convention.unit-testing.gradle.kts:2]: https://github.com/avito-tech/avito-android/pull/1131
